rust/actyx/certs/src/app_manifest.rs

```diff
@@ -30,31 +30,65 @@ impl AppManifestSignature {
     }
 }
 
-fn serialize_signature<S: Serializer>(sig: &AppManifestSignature, s: S) -> Result<S::Ok, S::Error> {
+fn serialize_signature<S: Serializer>(sig: &Option<AppManifestSignature>, s: S) -> Result<S::Ok, S::Error> {
     let bytes = serde_cbor::to_vec(sig).map_err(serde::ser::Error::custom)?;
     s.serialize_str(&base64::encode(bytes))
 }
 
-fn deserialize_signature<'de, D: Deserializer<'de>>(d: D) -> Result<AppManifestSignature, D::Error> {
-    let s = <String>::deserialize(d)?;
-    let data = base64::decode(s).map_err(|_| D::Error::custom("failed to base64 decode app manifest signature"))?;
-    serde_cbor::from_slice::<AppManifestSignature>(&data)
-        .map_err(|_| D::Error::custom("failed to deserialize to app manifest signature"))
+impl FromStr for AppManifestSignature {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let data = base64::decode(s).map_err(|_| "failed to base64 decode app manifest signature")?;
+        serde_cbor::from_slice::<AppManifestSignature>(&data)
+            .map_err(|_| "failed to deserialize to app manifest signature")
+    }
 }
 
-#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+fn deserialize_signature<'de, D: Deserializer<'de>>(deserializer: D) -> Result<Option<AppManifestSignature>, D::Error> {
+    let s = String::deserialize(deserializer)?;
```

Comment:
Will this cause an error if signature field does not exist (e.g. because if `skip_serializing_if` flag below).
I figured that the failing harness test was due to failing AppManifest deserialization in `src/auth/mod.rs, route function`

Reply:
NVM, I just realized there are 2 different `AppManifest` s